### PR TITLE
[SC] Add ToStringJsonConverter for UInt128/UInt256

### PR DESCRIPTION
**Pull request by quantumagi**

### CODE REVIEW

**Review by mrtpain — APPROVED** [review]
Looks good to me and working well 👍 nice work, thanks again! 😎

**Review by fassadlr — APPROVED** [review]

File: src/Stratis.Bitcoin.Features.SmartContracts/ApiLogDeserializer.cs [hunk 1]
```diff
@@ -55,6 +55,9 @@ public dynamic DeserializeLogData(byte[] bytes, Type type)
                 {
                     object fieldValue = this.primitiveSerializer.Deserialize(fieldType, fieldBytes);
 
+                    if (fieldType == typeof(UInt128) || fieldType == typeof(UInt256))
```
Comment (YakupIpek) [inline src/Stratis.Bitcoin.Features.SmartContracts/ApiLogDeserializer.cs:58]: You already fixed json deserialization issue by ToStringJsonConverter class so this code is not necessary.

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/ContractParametersContractResolver.cs [hunk 2]
```diff
@@ -23,6 +23,16 @@ protected override JsonContract CreateContract(Type objectType)
                 contract.Converter = new AddressJsonConverter(this.network);
             }
 
+            if (objectType == typeof(UInt128))
+            {
```
Comment (YakupIpek) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/ContractParametersContractResolver.cs:27]: These if statements should be if else statements

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/ToStringJsonConverter.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Stratis.Bitcoin.Features.SmartContracts.ReflectionExecutor
+{
+    public class ToStringJsonConverter<T> : JsonConverter
+    {
+        private readonly Network network;
+
+        public ToStringJsonConverter(Network network)
```
Comment (YakupIpek) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/ToStringJsonConverter.cs:12]: Network param is not needed for this class.

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/ToStringJsonConverter.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            JToken t = JToken.FromObject(value);
+
+            if (t.Type != JTokenType.Object)
+            {
+                t.WriteTo(writer);
+            }
+            else
+            {
+                JValue v = JValue.CreateString(((T)value).ToString());
```
Comment (YakupIpek) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/ToStringJsonConverter.cs:42]: Casting is not needed. ToString is overrided method.

**Review by YakupIpek — CHANGES_REQUESTED** [review]
Can you also add new types to Deserialize_Basic_Log_Success method.

File: src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/ToStringJsonConverter.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            throw new NotImplementedException("Unnecessary because CanRead is false. The type will skip the converter.");
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            JToken t = JToken.FromObject(value);
+
+            if (t.Type != JTokenType.Object)
```
Comment (YakupIpek) [inline src/Stratis.Bitcoin.Features.SmartContracts/ReflectionExecutor/ToStringJsonConverter.cs:36]: If section is not needed.

File: src/Stratis.Bitcoin.Features.SmartContracts/ApiLogDeserializer.cs (hunk 1 above)
Comment (mrtpain) [reply src/Stratis.Bitcoin.Features.SmartContracts/ApiLogDeserializer.cs:58]: This was necessary for logs to return correctly, without it, return values from a `local-call` would be correct but logs would still return `{}`.

**Pull request merged**